homeassistant/components/jellyfin/remote.py

```diff
@@ -0,0 +1,114 @@
+import time
+from typing import Any
+
+from homeassistant.components.remote import (
+    ATTR_DELAY_SECS,
+    ATTR_NUM_REPEATS,
+    DEFAULT_DELAY_SECS,
+    DEFAULT_NUM_REPEATS,
+    RemoteEntity,
+    RemoteEntityDescription,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import JellyfinConfigEntry
+from .const import DOMAIN
+from .coordinator import JellyfinDataUpdateCoordinator
+from .entity import JellyfinEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: JellyfinConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Jellyfin remote from a config entry."""
+    jellyfin_data = entry.runtime_data
+    coordinator = jellyfin_data.coordinators["sessions"]
+
+    @callback
+    def handle_coordinator_update() -> None:
+        """Add remote per session."""
+        entities: list[RemoteEntity] = []
+        for session_id, session_data in coordinator.data.items():
+            if (
+                session_id not in coordinator.remote_session_ids
+                and session_data["SupportsRemoteControl"]
+            ):
+                entity: RemoteEntity = JellyfinRemote(
+                    coordinator, session_id, session_data
+                )
+                coordinator.remote_session_ids.add(session_id)
+                entities.append(entity)
+        async_add_entities(entities)
+
+    handle_coordinator_update()
+
+    entry.async_on_unload(coordinator.async_add_listener(handle_coordinator_update))
+
+
+class JellyfinRemote(JellyfinEntity, RemoteEntity):
+    """Defines a Jellyfin remote entity."""
+
+    def __init__(
+        self,
+        coordinator: JellyfinDataUpdateCoordinator,
+        session_id: str,
+        session_data: dict[str, Any],
+    ) -> None:
+        """Initialize the Jellyfin Remote entity."""
+        super().__init__(
+            coordinator,
+            RemoteEntityDescription(
+                key=session_id,
+            ),
+        )
+
+        self.session_id = session_id
+        self.session_data: dict[str, Any] = session_data
+        self.device_id: str = session_data["DeviceId"]
+        self.device_name: str = session_data["DeviceName"]
+        self.client_name: str = session_data["Client"]
+        self.app_version: str = session_data["ApplicationVersion"]
+
+        self.capabilities: dict[str, Any] = session_data["Capabilities"]
+
+        if self.capabilities.get("SupportsPersistentIdentifier", False):
+            self._attr_device_info = DeviceInfo(
+                identifiers={(DOMAIN, self.device_id)},
+                manufacturer="Jellyfin",
+                model=self.client_name,
+                name=self.device_name,
+                sw_version=self.app_version,
+                via_device=(DOMAIN, coordinator.server_id),
+            )
```

Comment:
Can you elaborate what this entity does? As in, afaik a remote is usually a physical device, and Jellyfin is software, so is the remote virtual or did they introduce a new feature?

Reply:
Correct, this is a virtual remote for sending commands to clients. Does the linked doc PR clarify this? If not, let me know if there's anything there that can be improved. I'm implementing this as someone that's fairly familiar with Jellyfin so I might leave out information that I assume everyone knows (but doesn't).

https://github.com/home-assistant/home-assistant.io/pull/34871/files

Reply:
How many remotes can a server have? Or does this create a remote for every client?

Reply:
Correct, one per client.

Reply:
Do we already have a device per client?

Reply:
Yes, this uses the same logic as the media player entity for that.

Reply:
Would it make sense to maybe do a step backwards first and create a new base enttiy for the JellyfinClient? because now the base entity is for the server and we overwrite practically everything to make it a client entity. WDYT?

Reply:
I was considering that, I just went with minimal changes to other areas for the initial implementation. But I agree, I'll add that.

Reply:
I think it's simple enough to include here, see 5018ebef3ce5b6820e9fe1191c03e1cfa6af0184.